src/openrct2/actions/TileModifyAction.hpp

```diff
@@ -158,104 +162,110 @@ DEFINE_GAME_ACTION(TileModifyAction, GAME_COMMAND_MODIFY_TILE, GameActionResult)
                 break;
             }
             case TileModifyType::PathSetBroken:
             {
-                const auto elementIndex = _value1;
+                elementIndex = _value1;
                 const bool broken = _value2;
                 res = tile_inspector_path_set_broken(_loc, elementIndex, broken, isExecuting);
                 break;
             }
             case TileModifyType::PathToggleEdge:
             {
-                const auto elementIndex = _value1;
+                elementIndex = _value1;
                 const auto edgeIndex = _value2;
                 res = tile_inspector_path_toggle_edge(_loc, elementIndex, edgeIndex, isExecuting);
                 break;
             }
             case TileModifyType::EntranceMakeUsable:
             {
-                const auto elementIndex = _value1;
+                elementIndex = _value1;
                 res = tile_inspector_entrance_make_usable(_loc, elementIndex, isExecuting);
                 break;
             }
             case TileModifyType::WallSetSlope:
             {
-                const auto elementIndex = _value1;
+                elementIndex = _value1;
                 const auto slopeValue = _value2;
                 res = tile_inspector_wall_set_slope(_loc, elementIndex, slopeValue, isExecuting);
                 break;
             }
             case TileModifyType::WallSetAnimationFrame:
             {
-                const auto elementIndex = _value1;
+                elementIndex = _value1;
                 const auto animationFrameOffset = _value2;
                 res = tile_inspector_wall_animation_frame_offset(_loc, elementIndex, animationFrameOffset, isExecuting);
                 break;
             }
             case TileModifyType::TrackBaseHeightOffset:
             {
-                const auto elementIndex = _value1;
+                elementIndex = _value1;
                 const auto heightOffset = _value2;
                 res = tile_inspector_track_base_height_offset(_loc, elementIndex, heightOffset, isExecuting);
                 break;
             }
             case TileModifyType::TrackSetChainBlock:
             {
-                const auto elementIndex = _value1;
+                elementIndex = _value1;
                 const bool setChain = _value2;
                 res = tile_inspector_track_set_chain(_loc, elementIndex, true, setChain, isExecuting);
                 break;
             }
             case TileModifyType::TrackSetChain:
             {
-                const auto elementIndex = _value1;
+                elementIndex = _value1;
                 const bool setChain = _value2;
                 res = tile_inspector_track_set_chain(_loc, elementIndex, false, setChain, isExecuting);
                 break;
             }
+            case TileModifyType::TrackSetSpeed:
+                elementIndex = _value1;
+                assert(_value2 <= 0xFFFF);
```

Comment:
why check that is less than 0xFFFF when you then cast it to uint8_t. You should not have asserts in code like this either. What happens if a malicous individual sends a command to modify the track speed with a value > 0xFFFF, it crashes the server. Always validate safely and never throw an un caught error in action code.